test/functional/test_framework/util.py

```diff
@@ -55,6 +55,11 @@ def assert_greater_than(thing1, thing2):
         raise AssertionError("%s <= %s" % (str(thing1), str(thing2)))
 
 
+def assert_no_key(k, d):
+    if k in d:
+        raise AssertionError("%s in %s" % (str(k), str(d)))
```

Comment:
suggestion
```diff
-def assert_no_key(k, d):
-    if k in d:
-        raise AssertionError("%s in %s" % (str(k), str(d)))
+def assert_no_key(key, dictionary):
+    if key in dictionary:
+        raise AssertionError(f"Key '{key}' not expected to be found in {dictionary}")
```

which uses Python f-strings per current practice and would improve the error message from, for example

```
    raise AssertionError("%s in %s" % (str(k), str(d)))
AssertionError: size in {'loaded': True, 'size': 1, 'bytes': 219, 'usage': 1072, 'total_fee': Decimal('0.00004380'), 'maxmempool': 300000000, 'mempoolminfee': Decimal('0.00001000'), 'minrelaytxfee': Decimal('0.00001000'), 'unbroadcastcount': 1}
```
to
```
    raise AssertionError(f"Key '{key}' not expected to be found in {dictionary}")
AssertionError: Key 'size' not expected to be found in {'loaded': True, 'size': 1, 'bytes': 219, 'usage': 1072, 'total_fee': Decimal('0.00004380'), 'maxmempool': 300000000, 'mempoolminfee': Decimal('0.00001000'), 'minrelaytxfee': Decimal('0.00001000'), 'unbroadcastcount': 1}
```

Reply:
Nice, I didn't know about these f-string. Thanks.